source/fundamentals/aggregation/aggregation-expression-operations.txt

```diff
@@ -192,6 +192,156 @@ the Query API:
+
+   * - `switchOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#switchOn(java.util.function.Function)>`__
+
+       | **Typed Variants**:
+       | `switchArrayOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#switchArrayOn(java.util.function.Function)>`__
+       | `switchBooleanOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#switchBooleanOn(java.util.function.Function)>`__
+       | `switchDateOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDate.html#switchDateOn(java.util.function.Function)>`__
+       | `switchDocumentOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#switchDocumentOn(java.util.function.Function)>`__
+       | `switchIntegerOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#switchIntegerOn(java.util.function.Function)>`__
+       | `switchMapOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#switchMapOn(java.util.function.Function)>`__
+       | `switchNumberOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlNumber.html#switchNumberOn(java.util.function.Function)>`__
+       | `switchStringOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#switchStringOn(java.util.function.Function)>`__
+
+     - :manual:`$switch </reference/operator/aggregation/switch/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Suppose you have a collection which has a generic parameter used to store
+miscellaneous customer data. What you want to do with that data varies,
+depending on the type of data that is stored.
+
+To change what you are doing based on the data type of ``miscData``, use the
+``switchOn()`` method. If the data type is ``String``, just copy the data
+without altering it. If the data type is ``Boolean``, check the value with the
+``cond()`` method and return a string indicating whether the customer is a
+premium member. If the data type is ``Array``, get the first element by using
+the ``elementAt()`` method. If it is not one of these types, record a message
+by using ``defaults()``. Regardless of the type, use the ``project()`` method
+to output the result as ``result``.
+
+The following code shows the pipeline stage for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   Arrays.asList(project(fields(
+         computed("result",
+               current().getField("miscData").switchOn(field -> field
+                        .isString(s -> s)
+                        .isBoolean(b -> b.cond(
+                                 of("Customer is premium member"),
+                                 of("Customer is not premium member")))
+                        .isArray(a -> a.elementAt(0).asString())
+                        .defaults(d -> of("Not an anticipated type"))
+               ))
```

Comment:
There are two "patterns" where switch applies: 1) you have the same data, stored in different formats, and 2) you want to branch off on specific conditions (much like a typical switch, or a sequence of if-else statements).

This example is good in that it covers many important points for number 1, but I found it difficult to follow, I think because it seemed arbitrary or artificial. 

I think it could be improved by explaining (or implying / assuming) that, in the example, there was a "member" field that contained true or false indicating whether the customer had a premium membership, but then perhaps other membership levels were introduced (like gold and platinum), which were stored as strings, and evidently customers can now also have multiple memberships, the first (perhaps `.last`?) of which is their most recent.

```suggestion
         computed("membershipLevel",
               current().getField("member").switchOn(field -> field
                        .isString(s -> s)
                        .isBoolean(b -> b.cond(of("premium"), of("regular")))
                        .<MqlString>isArray(a -> a.last())
                        .defaults(d -> of("regular"))
               ))
```